tests/platforms/kubernetes/appmanager.go

```diff
@@ -196,6 +202,53 @@ func (m *AppManager) ValidiateSideCar() (bool, error) {
 	return true, nil
 }
 
+// DoPortForwarding performs port forwarding to access test apps in the cluster
+func (m *AppManager) DoPortForwarding() error {
+	podClient := m.client.Pods(m.namespace)
+	// Filter only 'testapp=appName' labeled Pods
+	podList, err := podClient.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", TestAppLabelKey, m.app.AppName),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	var podName string
+
+	for _, pod := range podList.Items {
+		podName = pod.Name
```

Comment:
is this port-forwarding to the first pod in the pods list ? Would it be good to pass the pod name in `DoPortFowarding()` ? So if no pod name passed, then we can pick the first pod.

Reply:
Or, we can port forward to the service instead of the pod.

Reply:
@yaron2 is it possible to port-forward to service ? I did not know it...

Reply:
> is this port-forwarding to the first pod in the pods list ? Would it be good to pass the pod name in `DoPortFowarding()` ? So if no pod name passed, then we can pick the first pod.

Thanks for review. I have now taking the podname in the request else pick the first matching pod name.

Reply:
> Or, we can port forward to the service instead of the pod.

thanks @yaron2  for review. Through kubectl it is possible to do port forward to the service but I made this change for pod forward only. I will expose/pass arg as a service forward later and open/enhancement issue for this change.